examples/vae/train_vae.py

```diff
@@ -20,60 +21,79 @@ def main():
                         help='Resume the optimization from snapshot')
     parser.add_argument('--gpu', '-g', default=-1, type=int,
                         help='GPU ID (negative value indicates CPU)')
-    parser.add_argument('--out', '-o', default='result',
+    parser.add_argument('--out', '-o', default='results',
                         help='Directory to output the result')
     parser.add_argument('--epoch', '-e', default=100, type=int,
                         help='number of epochs to learn')
-    parser.add_argument('--dimz', '-z', default=20, type=int,
+    parser.add_argument('--dim-z', '-z', default=20, type=int,
                         help='dimention of encoded vector')
-    parser.add_argument('--batchsize', '-b', type=int, default=100,
+    parser.add_argument('--dim-h', default=500, type=int,
+                        help='dimention of hidden layer')
+    parser.add_argument('--beta', default=1.0, type=float,
+                        help='Regularization coefficient for '
+                             'the second term of ELBO bound')
+    parser.add_argument('--k', '-k', default=1, type=int,
+                        help='Number of Monte Carlo samples used in '
+                             'encoded vector')
+    parser.add_argument('--batch-size', '-b', type=int, default=100,
                         help='learning minibatch size')
     parser.add_argument('--test', action='store_true',
                         help='Use tiny datasets for quick tests')
     args = parser.parse_args()
 
     print('GPU: {}'.format(args.gpu))
-    print('# dim z: {}'.format(args.dimz))
-    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('# dim z: {}'.format(args.dim_z))
+    print('# Minibatch-size: {}'.format(args.batch_size))
     print('# epoch: {}'.format(args.epoch))
     print('')
 
     # Prepare VAE model, defined in net.py
-    model = net.VAE(784, args.dimz, 500)
+    encoder = net.make_encoder(784, args.dim_z, args.dim_h)
+    decoder = net.make_decoder(784, args.dim_z, args.dim_h)
+    prior = net.make_prior(args.dim_z, device=args.gpu)
+    avg_elbo_loss = net.AvgELBOLoss(encoder, decoder, prior,
+                                    beta=args.beta, k=args.k)
 
     # Setup an optimizer
     optimizer = chainer.optimizers.Adam()
-    optimizer.setup(model)
+    optimizer.setup(avg_elbo_loss)
 
     # Initialize
     if args.initmodel:
-        chainer.serializers.load_npz(args.initmodel, model)
+        chainer.serializers.load_npz(args.initmodel, avg_elbo_loss)
 
     # Load the MNIST dataset
     train, test = chainer.datasets.get_mnist(withlabel=False)
+
+    # Binarize dataset
+    train[train >= 0.5] = 1.0
+    train[train < 0.5] = 0.0
```

Comment:
How about `train = (train >= 0.5).astype(np.float32)`?

Reply:
Thank you. It seems smarter than the previous way. I've changed it: 099a98c